### PR TITLE
Add available data length checking

### PR DESCRIPTION
**Pull request by Fr0sT-Brutal**
to avoid reading of garbage data

### CODE REVIEW

File: src/smartbuffer.ts [hunk 1]
```diff
@@ -749,6 +749,16 @@ class SmartBuffer {
         this.clear();
     }
 
+    /**
+     * Ensures that the internal Buffer is large enough to read data.
+     *
+     * @param length { Number } The length of the data that needs to be read.
+     */
+    ensureReadable(length) {
```
Comment (JoshGlazebrook) [inline src/smartbuffer.ts]: From a style standpoint, could you please refactor this to be a private method and add the number type to length.

```typescript
private ensureReadable(length: number) {
```

**Comment by JoshGlazebrook** [comment]
So if we're going to be adding bounds checking for reading, we probably want to add it for strings and buffers as well.

File: src/smartbuffer.ts (hunk 1 above)
Comment (Fr0sT-Brutal) [inline src/smartbuffer.ts]: Done! I'm totally new to TypeScript :)

**Comment by Fr0sT-Brutal** [comment]
Well, now they're being read as much as possible and someone could already count on it. I wasn't sure if such a significant behavior is suitable. My fix avoids an issue with reading a garbage data while buffer/string reading is a matter of preference so it's you to decide ;)

**Comment by JoshGlazebrook** [comment]
This was added and published into 3.0.0.

**Pull request merged**